lib/mayaUsd/ufe/UsdStageMap.cpp

```diff
@@ -158,44 +170,27 @@ MObject UsdStageMap::proxyShape(const Ufe::Path& path)
 
     if (iter == std::end(fPathToObject)) {
         // When we don't find an entry in the cache then we are in scenerio 1.
-        // MObjects stay valid even when re-parented or re-named, so we can
-        // scan through all the entries in the cache and validate that the current
-        // DAG path to the MObject matches the key Ufe::Path for the MObject in
-        // the cache. When the don't match, update fPathToObject so that the key and
-        // the MObject are in sync again.
-        auto pathToObject = fPathToObject;
-        for (const auto& entry : pathToObject) {
-            const auto& cachedPath = entry.first;
-            const auto& cachedObject = entry.second;
-            // Get the UFE path from the map value.
-            auto newPath = firstPath(cachedObject);
-            if (newPath != cachedPath) {
-                // Key is stale.  Remove it from our cache, and add the new entry.
-                auto count = fPathToObject.erase(cachedPath);
-                TF_AXIOM(count);
-                fPathToObject[newPath] = cachedObject;
-            }
-        }
+        rebuildCache();
```

Comment:
Isn't it possible to get into a state where we rebuilt the cache (because it was dirty) on line 140. Then you are forcing it to rebuild again here.

Reply:
Normally the 1st (conditional) rebuild would make the cache valid and not fail here, but there maybe cases were we rebuild twice. For example when accessing data that is really gone. In fact, currently nothing calls setDirty() in the whole code base... I think I could just get rid of the 1st conditional rebuild and let the code here do its thing, you are right.